### PR TITLE
distsql_test: fix TestPartitionTableIndexLookUpReader

### LINKED ISSUE #39516: flaky test TestPartitionTableIndexLookUpReader
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
        	Error:      	Not equal: 
        	            	expected: "[1 1]\n[2 2]\n[11 11]\n"
        	            	actual  : "[11 11]\n[12 12]\n[1 1]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	+[11 11]
        	            	+[12 12]
        	            	 [1 1]
        	            	-[2 2]
        	            	-[11 11]
        	            	 
        	Test:       	TestPartitionTableIndexLookUpReader
        	Messages:   	sql:select * from t where a between 1 and 15 limit 3, args:[]
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #39516 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/distsql_test.go [hunk 1]
```diff
@@ -316,9 +316,8 @@ func TestPartitionTableIndexLookUpReader(t *testing.T) {
 	tk.MustQuery("select * from t where a>=1 and a<15 order by a").Check(testkit.Rows("1 1", "2 2", "11 11", "12 12"))
 	tk.MustQuery("select * from t where a>=1 and a<15 order by a limit 1").Check(testkit.Rows("1 1"))
 	tk.MustQuery("select * from t where a>=1 and a<15 order by a limit 3").Check(testkit.Rows("1 1", "2 2", "11 11"))
-	tk.MustQuery("select * from t where a>=1 and a<15 limit 3").Check(testkit.Rows("1 1", "2 2", "11 11"))
```
Comment (Defined2014) [inline executor/distsql_test.go:319]: After add `order by`, it will same as L318, so delete it.

**Review by hawkingrei — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**